kafka-webview-ui/src/main/resources/application.yml

```diff
@@ -1,14 +1,95 @@
+
```

Comment:
Does it not make sense to have this use a profile as well to avoid having to redefine every property in this config?

Instead I'd like to be able to only surface the most relevant/common settings and hide away the complexities of configuring springboot.

Reply:
> Does it not make sense to have this use a profile as well to avoid having to redefine every property in this config?

Spring Profiles are intended to make easier the customisation (add, update) of specific application properties. We don't have to redefine every time all properties in each configurations. The example is `application-dev.yaml`:
- `server.servlet.session.persistent` is set to `true` instead of the default `false`
- `spring.jpa.show-sql = true` is added
Or did I understand wrong your question?

> Instead I'd like to be able to only surface the most relevant/common settings and hide away the complexities of configuring springboot.

We can hide Spring Boot complexity defining some easier environment variables. But I don't think that changing completely the way configurations are set is an easier way, taking away some complexity. Instead you have to figure out a new way, more difficult and not really well documented compared to the standard Spring Boot way.
Instead we can think about something like this:
```
spring:
  application:
    name: ${APP_NAME:"Kafka Webview"}
management:
  server:
    port: ${MANAGEMENT_PORT:9090}
```

Already done by you before for the `server.port`:
```
server:
  port: ${PORT:8080}
```

Reply:
The way I described above about defining easier environment variables, we can decide which configuration should be easy to be overridden and provide documentation for that. If someone wants to override something else deeper in Spring Boot, it's his/her duty to understand how.

Reply:
So what do you think about my last comment?
Can we define a set of easy-to-configure properties like suggested above and the rest will remain up to the user?